lib/Ix/Processor/JMAP.pm

```diff
@@ -108,6 +109,65 @@ sub _sanity_check_calls ($self, $calls, $arg) {
   return;
 }
 
+sub expand_backrefs ($self, $ctx, $arg) {
+  my @backref_keys = map {; s/^#// ? $_ : () } keys %$arg;
+
+  my sub ref_error ($desc) {
```

Comment:
Micro-opt - maybe - move this below the return immediately following since I think the common case is going to be not using backrefs

Reply:
I'll move it for clarity, but I don't think there's any optimization here.  A sub declaration is compile time, not runtime cost.

Reply:
Because it's a lexical sub (and even though it's not closing over anything) I wasn't sure if some sort of reinitialization/resetting was happening every time